packages/wrangler/src/bundle.ts

```diff
@@ -391,6 +397,52 @@ async function applyMultiWorkerFacade(
 	};
 }
 
+/**
+ * A middleware that makes forst party workers "work" in
+ * our dev environments. Is applied during wrangler dev
+ * when config.first_party_worker is true
+ */
+async function applyFirstPartyWorkerDevFacade(
+	entry: Entry,
+	tmpDirPath: string
+) {
+	if (entry.format !== "modules") {
+		throw new Error(
+			"First party workers must be in the modules format. See https://developers.cloudflare.com/workers/learning/migrating-to-module-workers/"
+		);
+	}
+
+	const targetPath = path.join(
+		tmpDirPath,
+		"first-party-worker-module-facade.entry.js"
+	);
+
+	await esbuild.build({
+		entryPoints: [
+			path.resolve(
+				__dirname,
+				entry.format === "modules"
```

Comment:
why do this if:
> First party workers must be in the modules format.
?

Reply:
whoops, forgot to remove the sw bits. nice catch!